riscos_toolbox/application.py

```diff
@@ -1,15 +1,48 @@
-from .events import EventHandler
+from .events import EventHandler, registered_wimp_events
+from consts import Wimp
 from . import initialise, run
 
 _application = None
 
 
+def _make_poll_flags(events):
```

Comment:
I suggest making this a global dictionary rather than a set of conditions, as it's largely data driven. eg:

```
wimp_event_masks = {
    Wimp.Null: Wimp.Poll.NullMask
    Wimp.RedrawWindow: Wimp.Poll.RedrawWindowRequestMask
    Wimp.PointerLeavingWindow: Wimp.Poll.PointerLeavingWindowMask
    Wimp.PointerEnteringWindow: Wimp.Poll.PointerEnteringWindowMask
    Wimp.MouseClick: Wimp.Poll.MouseClickMask
    Wimp.KeyPressed: Wimp.Poll.KeyPressedMask
    Wimp.LoseCaret: Wimp.Poll.LoseCaretMask
    Wimp.GainCartet: Wimp.Poll.GainCaretMask
    Wimp.PollwordNonZero: Wimp.Poll.PollWordNonZeroMask
}

...

for (event, mask) in wimp_event_masks.items():
    if event in events:
        flags |= mask
```

Reply:
Think it should be if event not in events: ?

Reply:
oops, yup!